tests/test_init.py

```diff
@@ -351,3 +351,54 @@ def test_zero_wires(self, init, sgntr):
         p = init(**sgntr)
 
         assert p.flatten().shape == (0,)
+
+
+    def test_particle_conserving_u1_init(self, tol):
+        """Test the functions 'particle_conserving_u1_uniform' and
+        'particle_conserving_u1_normal'."""
+
+        n_layers = 2
+        n_wires = 4
+
+        # check the shape
+        exp_shape = (n_layers, n_wires - 1, 2)
+        params = qml.init.particle_conserving_u1_uniform(n_layers, n_wires)
+        assert params.shape == exp_shape
+
+        params = qml.init.particle_conserving_u1_normal(n_layers, n_wires)
+        assert params.shape == exp_shape
+
+        # check deterministic output for a fixed seed
+        seed = 1975
+        p1 = qml.init.particle_conserving_u1_uniform(n_layers, n_wires, seed=seed)
+        p2 = qml.init.particle_conserving_u1_uniform(n_layers, n_wires, seed=seed)
+        assert np.allclose(p1, p2, atol=tol)
+
+        p1 = qml.init.particle_conserving_u1_normal(n_layers, n_wires, seed=seed)
+        p2 = qml.init.particle_conserving_u1_normal(n_layers, n_wires, seed=seed)
+        assert np.allclose(p1, p2, atol=tol)
+
+        # check that the output is different for different seeds
+        p1 = qml.init.particle_conserving_u1_uniform(n_layers, n_wires, seed=seed)
+        p2 = qml.init.particle_conserving_u1_uniform(n_layers, n_wires, seed=seed+1)
+        assert not np.allclose(p1, p2, atol=tol)
+
+        p1 = qml.init.particle_conserving_u1_normal(n_layers, n_wires, seed=seed)
+        p2 = qml.init.particle_conserving_u1_normal(n_layers, n_wires, seed=seed+1)
+        assert not np.allclose(p1, p2, atol=tol)
```

Comment:
I know I may be guilty of this test pattern...I'm not too sure about it now, because a) strictly speaking, the results could be the same for two seeds (although controlled here and highly unlikely), and b) we are kind of just testing that `seeding` works in general.

Maybe leave for now, but in the template overhaul we should think of better ways.